### DISCUSSION TITLE
WebHostBuilder and Autofac support

### DISCUSSION THREAD

**Discussion by shessane** [abpframework/abp#7223] [Q&A] [answered]
Hi,

The startup template comes with the new HostBuilder integration, but we still rely on the WebHostBuilder

```csharp
internal static IHostBuilder CreateHostBuilder(string[] args) =>
    Host.CreateDefaultBuilder(args)
    .ConfigureWebHostDefaults(webBuilder =>
    {
        webBuilder.UseStartup<Startup>();
    })
    .UseAutofac()
    .UseSerilog();
```

I wonder if anyone managed to get autofac integration with the old WebhostBuilder ?
I tried several ways to do that (register with UseAutofac inside the AddApplication, ConfigureServices with AddAutofac, use BuildServiceProviderFromFactory, tried with netcore3.1 templates...). I get the following exception : 

> System.InvalidOperationException
  HResult=0x80131509
  Message=Could not find singleton service: Microsoft.Extensions.Configuration.IConfiguration, Microsoft.Extensions.Configuration.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60
  Source=Volo.Abp.Core
  StackTrace:
   at Microsoft.Extensions.DependencyInjection.ServiceCollectionCommonExtensions.GetSingletonInstance[T](IServiceCollection services)

**Comment by maliming** [comment, marked as answer]
hi

Why can't you use Generic Host?

**Reply by shessane** [reply]
Hi,

Some of our Apis are hosted on an Azure Service Fabric cluster. The ServiceFabric SDK still needs an IWebHost to declare a listener : https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-services-communication-aspnetcore#kestrel-in-a-stateless-service

```csharp
return new KestrelCommunicationListener(serviceContext,
	endpointName, (url, listener) =>
	{
		return new WebHostBuilder()
			.UseKestrel()
			.Configure(....)
			.Build();
	});

```

Request to add the generic host is tracked here : microsoft/service-fabric-aspnetcore#48

**Reply by maliming** [reply]
hi

Autofac does not support this.

**Reply by shessane** [reply]
Thanks

**Reply by Luifeng0** [reply]
Hi,

Some of our Apis are hosted on an Azure Service Fabric cluster. The ServiceFabric SDK still needs an IWebHost to declare a listener : https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-services-communication-aspnetcore#kestrel-in-a-stateless-service
